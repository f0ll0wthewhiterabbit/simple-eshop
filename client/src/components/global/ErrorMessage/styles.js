import styled from 'styled-components'
import { Avatar, Typography } from '@material-ui/core'

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`

export const IconWrapper = styled(Avatar)`
  margin: ${props => props.theme.spacing(1)}px;
  background-color: ${props => props.theme.palette.secondary.main};
`

export const Heading = styled(Typography)`
  margin: ${props => props.theme.spacing(2)}px;
  font-family: 'Montserrat', sans-serif;
  color: #1e1e1e;
  font-size: 22px;
  font-weight: 500;
  line-height: 1.2;
`

export const Message = styled(Typography)`
  margin: ${props => props.theme.spacing(1)}px;
  font-family: 'Montserrat', sans-serif;
  color: #727272;
  font-size: 18px;
  font-weight: 400;
  line-height: 1.2;
`
