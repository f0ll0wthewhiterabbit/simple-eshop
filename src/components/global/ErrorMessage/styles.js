import styled from 'styled-components'
import { Avatar, Typography } from '@material-ui/core'

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`

const IconWrapper = styled(Avatar)`
  margin: ${props => props.theme.spacing(1)}px;
  background-color: ${props => props.theme.palette.secondary.main};
`

const Heading = styled(Typography)`
  margin: ${props => props.theme.spacing(1)}px;
  font-weight: 500;
`

const Message = styled(Typography)`
  margin: ${props => props.theme.spacing(1)}px;
  font-weight: 400;
`

export { Wrapper, IconWrapper, Heading, Message }
