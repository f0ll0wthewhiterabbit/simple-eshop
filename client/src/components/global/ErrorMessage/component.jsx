import React from 'react'
import PropTypes from 'prop-types'
import PriorityHighOutlinedIcon from '@material-ui/icons/PriorityHighOutlined'

import { Wrapper, IconWrapper, Heading, Message } from './styles'

const ErrorMessage = ({ children, title }) => {
  return (
    <Wrapper>
      <IconWrapper>
        <PriorityHighOutlinedIcon />
      </IconWrapper>
      {title && (
        <Heading variant="h5" component="h1">
          {title}
        </Heading>
      )}
      {children && <Message variant="h6">{children}</Message>}
    </Wrapper>
  )
}

ErrorMessage.defaultProps = {
  children: '',
  title: '',
}

ErrorMessage.propTypes = {
  children: PropTypes.string,
  title: PropTypes.string,
}

export default ErrorMessage
