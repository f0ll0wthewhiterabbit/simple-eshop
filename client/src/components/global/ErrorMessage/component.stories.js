import React from 'react'
import ErrorMessage from './component'

export default {
  title: 'ErrorMessage',
  component: ErrorMessage,
  excludeStories: /.*Data$/,
}

export const errorMessageData = {
  title: 'Test title',
  children: 'Test message',
}

export const empty = () => <ErrorMessage />
export const withTitle = () => <ErrorMessage title={errorMessageData.title} />
export const withMessage = () => <ErrorMessage>{errorMessageData.children}</ErrorMessage>
export const withTitleAndMessage = () => <ErrorMessage {...errorMessageData} />
