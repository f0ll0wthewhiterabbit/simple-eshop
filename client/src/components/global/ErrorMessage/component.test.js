import React from 'react'
import { shallow } from 'enzyme'

import ErrorMessage from './component'
import findByTestAttr from '../../../utils/findByTestAttr'

describe('ErrorMessage component', () => {
  let props
  let wrapper

  const generateWrapper = passedProps => {
    const initialProps = {
      children: '',
      title: '',
    }
    props = { ...initialProps, ...passedProps }

    return shallow(<ErrorMessage {...props} />)
  }

  beforeEach(() => {
    wrapper = generateWrapper()
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it(`shouldn't contain title`, () => {
    const title = findByTestAttr(wrapper, 'heading')

    expect(title).toHaveLength(0)
  })

  it(`shouldn't contain message`, () => {
    const message = findByTestAttr(wrapper, 'message')

    expect(message).toHaveLength(0)
  })

  it(`should contain title passed from props`, () => {
    const testTitle = 'Test Title'
    wrapper = generateWrapper({ title: testTitle })
    const title = findByTestAttr(wrapper, 'heading')

    expect(title).toHaveLength(1)
    expect(title.text()).toBe(testTitle)
  })

  it(`should contain message passed as children`, () => {
    const testMessage = 'Test Message'
    wrapper = generateWrapper({ children: testMessage })
    const message = findByTestAttr(wrapper, 'message')

    expect(message).toHaveLength(1)
    expect(message.text()).toBe(testMessage)
  })
})
