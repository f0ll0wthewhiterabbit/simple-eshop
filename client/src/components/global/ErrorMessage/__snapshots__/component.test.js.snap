// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage component should render correctly 1`] = `
<styled.div>
  <Styled(WithStyles(ForwardRef(Avatar)))>
    <PriorityHighOutlinedIcon />
  </Styled(WithStyles(ForwardRef(Avatar)))>
</styled.div>
`;
